test/functional/runner/matcher.js

```diff
@@ -34,6 +42,37 @@ function generateMatchAndDiffSpecialCase(key, expectedObj, actualObj, metadata)
     };
   }
 
+  if (typeof expected === 'object' && Object.keys(expected)[0] === '$$type') {
```

Comment:
nit: break this part out into a sub-function with a cool name so it is clear what this special case is.